lib/ood_core/batch_connect/templates/vnccontainer.rb

```diff
@@ -0,0 +1,248 @@
+        # @option context [#to_s] :geometry ("") resolution of vnc display (not
+        #   set if blank or `:vnc_args` is set) (assumes you don't modify
+        #   `:before_script`)
+        # @option context [#to_s] :dpi ("") dpi of vnc display (not set if
+        #   blank or `:vnc_args` is set) (assumes you don't modify
+        #   `:before_script`)
+        # @option context [#to_s] :fonts ("") command delimited list of fonts
+        #   available in vnc display (not set if blank or `:vnc_args` is set)
+        #   (assumes you don't modify `:before_script`)
+        # @option context [#to_s] :idle ("") timeout vnc server if no
+        #   connection in this amount of time in seconds (not set if blank or
+        #   `:vnc_args` is set) (assumes you don't modify `:before_script`)
+        # @option context [#to_s] :extra_args ("") any extra arguments used
+        #   when initializing the vnc server process (not set if blank or
+        #   `:vnc_args` is set) (assumes you don't modify `:before_script`)
+        # @option context [#to_s] :vnc_clean ("...") script used to clean up
+        #   any active vnc sessions (assumes you don't modify `:before_script`
+        #   or `:clean_script`)
+        # @option context [#to_s] :container_path ("vnc_container.sif") the path
+        #   to the container with VNC
+        # @option context [#to_s] :container_bindpath ("") paths to bind into
+        #   the container with VNC
+        # @option context [#to_s] :container_module ("singularity") the module
+        #   that loads Singularity or Apptainer with Lmod. Supports versions (i.e.
+        #   apptainer/1.10). If Singularity or Apptainer are installed at a 
+        #   system level (i.e., no module loaded to activate), set this to an
+        #   empty string.
+        # @option context [#to_s] :container_command ("singularity") the 
+        #   singularity or apptainer execution command
+        # @option context [#to_s] :instance_name ("desktop") a name for the
+        #   apptainer instance
+        # @see Template
+        def initialize(context = {})
+          super
+        end
+
+        private
+          # We need to know the VNC and websockify connection information
+          def conn_params
+            (super + [:display, :websocket, :spassword, :instance_name]).uniq
+          end
+
+          # Before running the main script, start up a VNC server and record
+          # the connection information
+          def before_script
+            container_path = context.fetch(:container_path, "vnc_container.sif").to_s
+            container_bindpath = context.fetch(:container_bindpath, "").to_s
+
+            <<-EOT.gsub(/^ {14}/, "")
+
+              # Load #{container_module}
+              echo "Loading #{container_module}..."
+              module load #{container_module}
+              export #{container_command.upcase}_BINDPATH="#{container_bindpath}"
+              export INSTANCE_NAME="#{instance_name}"
```

Comment:
I believe this isn't passing through to `connection.yml` because bash environment variables are case sensitive.

```suggestion
              export INSTANCE_NAME="#{instance_name}"
              export instance_name="#{instance_name}"
```